api/apps/api/test/access-control/project-acl-delete.e2e-spec.ts

```diff
@@ -0,0 +1,38 @@
+import { FixtureType } from '@marxan/utils/tests/fixture-type';
+import { getFixtures } from './project-acl.fixtures';
+
+let fixtures: FixtureType<typeof getFixtures>;
+
+beforeEach(async () => {
+  fixtures = await getFixtures();
+});
+afterEach(async () => {
+  await fixtures?.userRoleCleanup();
+  await fixtures?.cleanup();
+});
+
+test(`delete every type of user from the project as owner`, async () => {
+  const projectId = await fixtures.GivenProjectExistsAndHasUsers();
+  let revokeResponse = await fixtures.WhenRevokingAccessToViewerFromProjectAsOwner(
+    projectId,
+  );
+  fixtures.ThenNoContentIsReturned(revokeResponse);
+  revokeResponse = await fixtures.WhenRevokingAccessToContributorFromProjectAsOwner(
+    projectId,
+  );
+  fixtures.ThenNoContentIsReturned(revokeResponse);
+  revokeResponse = await fixtures.WhenRevokingAccessToOwnerFromProjectAsOwner(
+    projectId,
+  );
+  fixtures.ThenNoContentIsReturned(revokeResponse);
+  const response = await fixtures.WhenGettingProjectUsersAsOwner(projectId);
+  fixtures.ThenSingleOwnerUserInProjectIsReturned(response);
```

Comment:
in general GWT syntax approach, it is advised to keep G -> W -> T order - this is, without mixing the action (When) with expectation (Then).

We could either group `When`s together or split the actions (as single action would be rather expected). In this case I think it is fine to just group them.

Reply:
I think I didn't understand this well. You mean grouping like I did here (all under one test) or grouping every endpoint request in a single `When` method?

Would not make that impossible to check the `ThenNoContentReturned` or also checking those inside the When Method and only using the last `Then` for the latest endpoint request the `When` method returns?

Reply:
@rubvalave in this very case I meant:

```
 await fixtures.GivenProjectExistsAndHasUsers();

 await fixtures.WhenRevokingAccessToViewerFromProjectAsOwner(...)
await fixtures.WhenRevokingAccessToOwnerFromProjectAsOwner(...)
//...

await fixtures.Then.....
```